### PR TITLE
DevDocs: Fix the Sidebar to highlight the currently selected menu

### PR DESCRIPTION
**Pull request by afsheenirani**
## Background
The DevDocs sidebar does not show the sidebar menu item that is currently selected.

## Testing
1. Launch the [devdocs](http://calypso.localhost:3000/devdocs?) link.
2. Click each menu item to make sure it navigates to the right destination and correctly highlights as being selected.

 - [Search](http://calypso.localhost:3000/devdocs?)
 - [The Calypso Guide](http://calypso.localhost:3000/devdocs/docs/guide/index.md)
 - [Contributing](http://calypso.localhost:3000/devdocs/CONTRIBUTING.md)
 - [UI Components](http://calypso.localhost:3000/devdocs/design)
 - [App Components](http://calypso.localhost:3000/devdocs/app-components)
 - [Typography](http://calypso.localhost:3000/devdocs/design/typography)
 - [Icons](http://calypso.localhost:3000/devdocs/docs/icons.md)

## Reviewers
@lancewillett @nb 

## Issues
Closes [DevDocs: sidebar navigation should highlight the selected element](https://github.com/Automattic/wp-calypso/issues/1422)

### CODE REVIEW

File: client/devdocs/controller.js [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
-	ReduxProvider = require( 'react-redux' ).Provider,
-	setSection = require( 'state/ui/actions' ).setSection,
-	EmptyContent = require( 'components/empty-content' );
+import ReactDom from 'react-dom';
+import React from 'react';
+import qs from 'qs';
+import debounce from 'lodash/debounce';
+import page from 'page';
+import { Provider } from 'react-redux';
+import { setSection } from 'state/ui/actions';
```
Comment (mtias) [inline client/devdocs/controller.js]: Not new, but this one and `EmptyContent` should be in the internal dependencies block.

File: client/devdocs/sidebar.jsx [hunk 2]
```diff
@@ -24,53 +24,60 @@ export default React.createClass( {
...
-						</li>
-						<li className="devdocs__navigation-item">
-							<Gridicon icon="pencil" />
-							<a className="devdocs__sidebar-item" href="/devdocs/CONTRIBUTING.md">
-								Contributing
-							</a>
-						</li>
+						<SidebarItem
+							className={ 'devdocs__navigation-item' }
+							icon={ 'search' }
```
Comment (mtias) [inline client/devdocs/sidebar.jsx]: All these could be just passed as direct string attributes: `icon="search"`.

**Comment by afsheenirani** [comment]
@mtias - Thanks for the review. Good spot on the dependencies. I also made the change to the SidebarItem attributes to make them regular strings.

**Comment by afsheenirani** [comment]
@mtias - Let me know if there is any further feedback. Thanks.

File: client/devdocs/controller.js [hunk 3]
```diff
@@ -1,43 +1,45 @@
...
-	page = require( 'page' ),
-	ReduxProvider = require( 'react-redux' ).Provider,
-	setSection = require( 'state/ui/actions' ).setSection,
-	EmptyContent = require( 'components/empty-content' );
+import ReactDom from 'react-dom';
+import React from 'react';
+import qs from 'qs';
+import debounce from 'lodash/debounce';
+import page from 'page';
+import { Provider } from 'react-redux';
```
Comment (aduth) [inline client/devdocs/controller.js]: Was there a reason you chose to revert to naming this `Provider` ? I don't feel strongly either way, but had originally worried that `Provider` was less easy to understand at a glance than `ReduxProvider`.

As an FYI, you can rename ES6 imports using `as`:

```js
import { Provider as ReduxProvider } from 'react-redux';
```

**Comment by aduth** [comment]
Aside from my one question, this looks and works quite well. :+1: 

Before this is merged, would you mind squashing your commits? See [guide](https://github.com/Automattic/wp-calypso/blob/master/CONTRIBUTING.md#lifecycle-of-a-pull-request) and [tutorial](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) for more information.

```bash
$ git remote add upstream git@github.com:Automattic/wp-calypso.git
$ git fetch upstream
$ git rebase -i upstream/master
$ # Change prefix for all but first item to "fixup", then save
$ git push -f origin/add/devdocs-sidebar-selected
```

File: client/devdocs/controller.js (hunk 3 above)
Comment (afsheenirani) [inline client/devdocs/controller.js]: I essentially, wanted to keep the naming consistent i.e., name of imported module should tie out to name of exported module (as is the convention being followed across the codebase). This will also avoid any funky scenarios where (say) a new ReduxProvider module gets exported in 'react-redux'.

I recently learnt of the `import as` construct, so will use that as you have suggested as well, since I agree that the `ReduxProvider` name is much better.

**Comment by afsheenirani** [comment]
@aduth - Took me some figuring out, but I managed to squash the commits. I believe this is in a good state now. Also, retested to make sure there are no issues.

**Comment by aduth** [comment]
Thanks @afsheenirani , it looks great. I'll plan to merge shortly.

Were there anything in particular about the rebase which was difficult to work through? Since this is common to our workflow, it's important that we iron out or clarify any confusing points.

**Comment by afsheenirani** [comment]
@aduth - I think the instructions you gave me were pretty much "on the dot"...though, the last command should have been (I think - at least that's what worked for me):
>$ git push -f origin add/devdocs-sidebar-selected
> and not...
>$ git push -f origin/add/devdocs-sidebar-selected

But that was not the issue (I figured the above was probably a typo), I faced a combination of issues because:
1. I'm pretty new to Git. :)
2. I was getting merge conflicts when doing a rebase.
3. Having extra 'incorrect' commits on my local master that I had to do a hard reset on
4. Also had to sync my fork 'origin' with upstream.

So in summary - it was me ;-)

One quick question though - I use Sublime Text 3 with the sublime-git plugin, this doesn't have alot of the features that git supports (is there a particular setup you would recommend? - something that provides alot of the functionality to (say) squash commits?

**Comment by mtias** [comment]
Thanks for this @afsheenirani, it's a nice improvement.

**Comment by mtias** [comment]
I just use the terminal for git commands myself.

**Comment by aduth** [comment]
@afsheenirani : Ah, sorry for the typo, you're correct. I use SublimeText 3 as well for editing, though switch to Terminal for resolving branch conflicts. I learned just enough Vim to get me through the interactive rebase prompt :wink: 

It's been on my list for a while to put together a screencast for this exact flow, so if/when that comes to fruition, it'll hopefully clear up some of these concerns.

**Comment by afsheenirani** [comment]
Thanks @mtias and @aduth! Appreciate your insight and feedback on reviews as well.
